tests/test_cacheable_pdf.py

```diff
@@ -0,0 +1,132 @@
+import zfit
+import zfit.z.numpy as znp
+import tensorflow as tf
+
+from zfit import supports
+from zfit.models.cache import CacheablePDF
+from zfit.util import ztyping
+
+
+class TestPDF(zfit.pdf.ZPDF):
+    _PARAMS = ["alpha"]
+
+    def __init__(self, obs: ztyping.ObsTypeInput, **params):
+        super().__init__(obs, **params)
+        self.pdf_call_counter = tf.Variable(0.0)
+        self.integrate_call_counter = tf.Variable(0.0)
+        _mu = zfit.Parameter("mu", 1.0, -5, 5)
+        _sigma = zfit.Parameter("sigma", 1, 0, 10)
+        _obs = zfit.Space("x", limits=[-5.0, 5.0])
+        self._gauss = zfit.pdf.Gauss(mu=_mu, sigma=_sigma, obs=_obs)
+
+    @supports(norm="space")
+    def _pdf(self, x, norm, *, norm_range=None):
+        self.pdf_call_counter.assign_add(1.0)
+        return self._gauss.pdf(x)
+
+    @supports(norm="space")
+    def _integrate(self, limits, norm, options=None):
+        self.integrate_call_counter.assign_add(1.0)
+        return self._gauss.integrate(limits)
+
+
+def test_cached_pdf_equals_pdf_without_cache():
+    mu = zfit.Parameter("mu", 1.0, -5, 5)
+    sigma = zfit.Parameter("sigma", 1, 0, 10)
+    obs = zfit.Space("x", limits=[-5.0, 5.0])
+    gauss = zfit.pdf.Gauss(mu=mu, sigma=sigma, obs=obs)
+    cached_gauss = CacheablePDF(gauss)
+    x = znp.linspace(-5, 5, 500)
+
+    assert tf.math.reduce_all(tf.equal(gauss.pdf(x), cached_gauss.pdf(x)))
```

Comment:
```suggestion
    assert tf.math.reduce_all(tf.equal(gauss.pdf(x), cached_gauss.pdf(x)))
```
just to make sure the cache is used